packages/host/app/components/operator-mode/card-inheritance-panel.gts

```diff
@@ -10,88 +10,118 @@ import {
-            @realmInfo={{@realmInfo}}
-            @realmIconURL={{@realmIconURL}}
-            @onSelectDefinition={{this.updateCodePath}}
-            @url={{t.module}}
-          />
-        {{/each}}
+      {{#if this.isLoading}}
+        <div>Loading...</div>
       {{else}}
-        {{! Module case when visting, eg author.gts }}
-        {{#each this.adoptionChainTypes as |t|}}
+        {{#if @cardInstance}}
+          {{! JSON case when visting, eg Author/1.json }}
+          <h3>Inheritance Panel</h3>
+          <div class='inheritance-chain'>
+            <InstanceDefinitionContainer
+              @name={{@cardInstance.title}}
+              @fileExtension='.JSON'
+              @realmInfo={{@realmInfo}}
+              @infoText={{this.lastModified}}
+              @actions={{array
+                (hash label='Delete' handler=@delete icon='icon-trash')
+              }}
+            />
+            <div>Adopts from</div>
+
+            <ClickableModuleDefinitionContainer
+              @name={{this.cardType.type.displayName}}
+              @fileExtension={{this.cardType.type.moduleMeta.extension}}
+              @realmInfo={{this.cardType.type.moduleMeta.realmInfo}}
+              @onSelectDefinition={{this.updateCodePath}}
+              @url={{this.cardType.type.module}}
+            />
+          </div>
+        {{else}}
+          {{! Module case when visting, eg author.gts }}
+          <h3>In This File</h3>
+          {{#each this.elementsInFile as |el|}}
+            <div
+              class='inheritance-chain
+                {{if (@adoptionChainManager.isSelected el) "selected"}}'
+            >
+              <div>{{this.getCardTypeDisplayName el.card}}</div>
+              <button {{on 'click' (fn this.select el)}}>Select</button>
+            </div>
+          {{/each}}
+          <h3>Inheritance Panel</h3>
           <ModuleDefinitionContainer
-            @name={{t.displayName}}
-            @fileExtension={{this.fileExtension}}
-            @realmInfo={{@realmInfo}}
-            @realmIconURL={{@realmIconURL}}
+            @name={{this.cardType.type.displayName}}
+            @fileExtension={{this.cardType.type.moduleMeta.extension}}
+            @realmInfo={{this.cardType.type.moduleMeta.realmInfo}}
             @isActive={{true}}
             @actions={{array
               (hash label='Delete' handler=@delete icon='icon-trash')
             }}
           />
-          <div>Inherits from</div>
-          <ClickableModuleDefinitionContainer
-            @name={{t.super.displayName}}
-            @fileExtension={{this.fileExtension}}
-            @realmInfo={{@realmInfo}}
-            @realmIconURL={{@realmIconURL}}
-            @onSelectDefinition={{this.updateCodePath}}
-            @url={{t.super.module}}
-          />
-        {{/each}}
+          {{#if this.cardType.type.super}}
+            <div>Inherits from</div>
+            <ClickableModuleDefinitionContainer
```

Comment:
I'm having some difficulties understanding why we have

1. ModuleDefinitionContainer
2. ClickableModuleDefinitionContainer

I am wondering why can't we just have a single `ModuleDefinitionContainer` which has conditional behaviour based on an argument.

Reply:
Hmmm well. I feel like there were confusing states.

I think the reason ClickableModuleDefinitionContainer is there is so that it doesn't require you to pass any actions down since we don't intend to display any actions as buttons (like delete or inherit etc). It wud be confusing to show buttons when the entire defintion pill is clickable. 

I am open to renaming these stuff if you have suggestions.  Or if there is a preferred design of using <Clickable> overlay.